app/dashboard/src/hooks/syncRefHooks.ts

```diff
@@ -14,9 +14,7 @@ export function useSyncRef<T>(value: T): Readonly<React.MutableRefObject<T>> {
 
   // Update the ref value whenever the provided value changes
   // Refs shall never change during the render phase, so we use `useEffect` here.
-  React.useEffect(() => {
-    ref.current = value
-  })
+  ref.current = value
```

Comment:
is this intentional? if so then the comment no longer applies

Reply:
Yes, even though the react core team doesn't recommend setting ref values during the render (it might lead to deoptimizations), the reasoning behind this is:

1. We want to make `useEventCallback` behave the same way as `const x = () => {}` or `useCallback` but have a stable reference.
2. React components shall be `idempotent` by default, and I don't see violations here.

Reply:
(in that case the comment should be removed - or replaced with an explanation why we want to explicitly set this during the render)